### ISSUE TITLE
Support for multi-language support [feature-request]

### ISSUE THREAD

**Issue by jevonearth** [spf13/hugo#477]
I would like to use hugo for a user documentation website. The base language will be English, and from the base language I will have it translated to additional languages, including right-to-left languages.

An example from the middleman project can be seen here: http://middlemanapp.com/advanced/localization/

**Comment by jmcclelland** [comment]
I'm really impressed with the flexibility of hugo when it comes to making a multi-lingual site. I've implemented my site according to the [hugo docs on mult-lingual sites](http://gohugo.io/tutorials/create-a-multilingual-site/) and it is working like a charm.

In my opinion, there is only one missing feature. Here is the problem: When creating layouts, instead of writing words and sentences as you want them to appear in the primary language, you have to insert a variable wrapped in a relatively convoluted syntax (e.g. ```{{ index ( index $.Site.Data.translations  $.Site.Params.locale ) "somestring" }}``). Then you have to manually add the variable to each of your language files.

It would be more convenient if I could 

 * Put in my layout file
```
{{ translate "Multi-lingual sites in Hugo are easy to create!" }}
```
 * Run xgettext on all my layout files to build a PO file. 
 * Edit all PO files with the proper translations (or send them to the translators)
 * Configure each config_LL.toml file with the path to the PO file for that language
 * Have hugo automatically insert the right text

**Comment by abourget** [comment]
or at least have a quick `{{ i18n "somestring" }}` that would take `Site.Data.translations`and `Site.Params.locale` as conventions..

**Comment by abourget** [comment]
I'd also really love Hugo not to require multiple domains.. and I'm looking for a way to have a "French -> English" link on each page, that links directly to the other language's version.

I was thinking of providing a map like `.Translations` on the `Page` variable.. in which we'd have something like `es -> URL` or an object on which we could call `ref` or `relref`... this way we could generate those links on-the-fly..

Maybe that'd require a convention for naming the content files.. like `my_blog_post.md` and `my_blog_post.es.md` .. and we'd have a `default_translation_lang` somewhere.. that'd be really neat.

How involved would it be for a newcomer to Hugo (but quite literate in Go) to implement such things ?

**Comment by abourget** [comment]
I submitted a PR to work on this here https://github.com/spf13/hugo/pull/1734

**Comment by tnolet** [comment]
Any updates on this? i18n support is probably a very basic feature for most semi-serious CMS-type users.

**Comment by fedelibre** [comment]
@tnolet follow the PR #1734 
It was about to be included in 0.16, but then Hugo team decided to take more time to test it.
You may want to try to use that patch in a project and give feedback.
